src/Versity.Products/External/Presentation/Extensions/ProgramExtensions.cs

```diff
@@ -0,0 +1,142 @@
+        builder.Services.AddJwtAuthentication(builder.Configuration);
+        builder.Services.AddControllers();
+        builder.Services.AddEndpointsApiExplorer();
+        builder.Services.AddSwaggerGen(options =>
+        {
+            options.AddSecurityDefinition("oauth2", new OpenApiSecurityScheme()
+            {
+                In = ParameterLocation.Header,
+                Name = "Authorization",
+                Type = SecuritySchemeType.ApiKey
+            });
+        });
+
+        builder.Services.AddCors(options => options.AddPolicy("AllowAll", policy =>
+        {
+            policy.AllowAnyHeader();
+            policy.AllowAnyMethod();
+            policy.AllowAnyOrigin();
+        }));
+        
+        builder.Host.UseSerilog((context, configuration) =>
+        {
+            var environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
+
+            if (string.IsNullOrEmpty(environment))
+                throw new NullReferenceException("ASPNETCORE_ENVIRONMENT variable is empty!");
+
+            var configurationRoot = new ConfigurationBuilder()
+                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{environment}.json", optional: true)
+                .Build();
+
+            configuration
+                .Enrich.FromLogContext()
+                .Enrich.WithMachineName()
+                .Enrich.WithExceptionDetails()
+                .WriteTo.Console()
+                .WriteTo.Elasticsearch(ConfigureElasticSink(configurationRoot, environment))
+                .Enrich.WithProperty("Environment", context.HostingEnvironment.EnvironmentName)
+                .ReadFrom.Configuration(context.Configuration);
+        });
+
+        builder.Services.AddDataProtection().UseCryptographicAlgorithms(new AuthenticatedEncryptorConfiguration
+        {
+            EncryptionAlgorithm = EncryptionAlgorithm.AES_256_CBC,
+            ValidationAlgorithm = ValidationAlgorithm.HMACSHA256
+        });
+
+        builder.Services.Configure<IISServerOptions>(options =>
+        {
+            options.AutomaticAuthentication = false;
+        });
+
+        return builder;
+    }
+    
+    public static WebApplication ConfigureApplication(this WebApplication app)
+    {
+        if (app.Environment.IsDevelopment())
+        {
+            app.UseExceptionHandler("/error-development");
+            app.UseSwagger();
+            app.UseSwaggerUI();
+        }
+        else
+        {
+            app.UseExceptionHandler("/error");
+        }
+
+        app.UseSerilogRequestLogging();
+        app.UseHttpsRedirection();
+        app.UseAuthentication();
+        app.UseAuthorization();
+        app.UseCors("AllowAll");
+        app.MapControllers();
+        
+        return app;
+    }
+    
+    private static IServiceCollection AddJwtAuthentication(this IServiceCollection serviceCollection, IConfiguration configuration)
```

Comment:
Extension method

Reply:
I didn't quite understand what you meant here, but I put the method data in a [separate class and file](https://github.com/jejikeh/versity/pull/17/files#diff-53bf72a090c03bb959c8336bb56b8f0c42d668dc9af9f66b9a9edddf0508c40f)